### PR TITLE
Added new tests for Gauss-Jordan elimination.

### PR DESCRIPTION
**Pull request by 1fisedi**

### CODE REVIEW

File: Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            var input = new double[,] { { 0, 0, 0 }, { 0, 0, 0 } };
+
+            // Act
+            var result = solver.Solve(input);
+
+            // Assert
+            Assert.IsFalse(result);
+        }
+    }
+}
```
Comment (siriak) [inline Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs]:
```suggestion
}

```

File: Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /// Class for testing Gauss-Jordan Elimination Algorithm.
+    /// </summary>
+    public static class GaussJordanEliminationTests
+    {
+        [Test]
+        public static void SolveMatrixThrowsException()
+        {
+            // Arrange
+            var solver = new GaussJordanElimination();
+
```
Comment (siriak) [inline Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs]:
```suggestion
```

File: Algorithms.Tests/Encoders/HillEnconderTests.cs [hunk 3]
```diff
@@ -15,7 +16,8 @@ public void DecodedStringIsTheSame()
             var encoder = new HillEncoder();
             var random = new Randomizer();
             var message = random.GetString();
-            var key = new double[,] { { 2, 4, 5 }, { 9, 2, 1 }, { 3, 17, 7 } };
+
+            var key = new double[,] { { 0, 4, 5 }, { 9, 2, -1 }, { 3, 17, 7 } };
```
Comment (siriak) [inline Algorithms.Tests/Encoders/HillEnconderTests.cs]: Why did you change the key, was it incorrect in some way?

File: Algorithms.Tests/Encoders/HillEnconderTests.cs [hunk 4]
```diff
@@ -15,7 +16,8 @@ public void DecodedStringIsTheSame()
             var encoder = new HillEncoder();
             var random = new Randomizer();
             var message = random.GetString();
-            var key = new double[,] { { 2, 4, 5 }, { 9, 2, 1 }, { 3, 17, 7 } };
+
```
Comment (siriak) [inline Algorithms.Tests/Encoders/HillEnconderTests.cs]:
```suggestion
```

File: Algorithms.Tests/Encoders/HillEnconderTests.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
 ﻿using Algorithms.Encoders;
+
```
Comment (siriak) [inline Algorithms.Tests/Encoders/HillEnconderTests.cs]:
```suggestion
```

File: Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+namespace Algorithms.Tests.Numeric
+{
+    /// <summary>
+    /// Class for testing Gauss-Jordan Elimination Algorithm.
+    /// </summary>
+    public static class GaussJordanEliminationTests
+    {
+        [Test]
+        public static void SolveMatrixThrowsException()
```
Comment (siriak) [inline Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs]: From the name of the test it's not clear **why** solver should fail, what is wrong with the input.

File: Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            // Assert
+            _ = Assert.Throws<ArgumentException>(Act);
+        }
+
+        [Test]
+        public static void UnableToSolveSingularMatrix()
+        {
+            // Arrange
+            var solver = new GaussJordanElimination();
+
```
Comment (siriak) [inline Algorithms.Tests/Numeric/GaussJordanEliminationTests.cs]:
```suggestion
```

File: Algorithms.Tests/Encoders/HillEnconderTests.cs (hunk 3 above)
Comment (1fisedi) [reply Algorithms.Tests/Encoders/HillEnconderTests.cs]: It increases the coverage of the test by including zeros and negative numbers. The alternative is to have multiple tests.
